crates/integration-tests/src/json_schema.rs

```diff
@@ -8,9 +8,43 @@ use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 use typify::TypeSpaceSettings;
 
-#[derive(Debug, JsonSchema)]
+#[derive(Debug)]
 enum MyKind {}
 
+impl JsonSchema for MyKind {
+    fn schema_name() -> String {
+        "MyKind".to_string()
+    }
+
+    fn is_referenceable() -> bool {
+        false
+    }
+
+    fn json_schema(_: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        schemars::schema::SchemaObject {
+            subschemas: Some(
+                schemars::schema::SubschemaValidation {
+                    not: Some(schemars::schema::Schema::Bool(true).into()),
+                    ..Default::default()
+                }
+                .into(),
+            ),
+            extensions: [(
+                "x-rust-type".to_string(),
+                serde_json::json!({
+                    "crate": "my-crate",
+                    "version": "0.1.0",
+                    "path": "my_crate::MyKind",
+                }),
+            )]
+            .into_iter()
+            .collect(),
+            ..Default::default()
+        }
+        .into()
+    }
+}
```

Comment:
This makes sense. Is there a plan to make this easier to define? Wonder if we should revive @hawkw's proc macro.

Reply:
That's something I wanted to talk about. I could imagine a derive macro, but it might be less rope for consumers to instead export something like the `impl_typed_uuid_kind` macro you suggest people write. As you noted, the derive macro requires a few inputs; this change would require more (crate name, path to the type)--I think that as derive macros grow mandatory attributes they start feeling like they should be function-style macros instead.